[\#2202 Issue](https://github.com/rear/rear/issues/2202) `closed`: 2.4 CIFS Mount fail
======================================================================================

**Labels**: `support / question`, `fixed / solved / done`

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) opened issue at [2019-08-06 22:15](https://github.com/rear/rear/issues/2202):

#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue  
(quick response is not guaranteed with free support):

-   ReaR version ("/usr/sbin/rear -V"):  
    2.4 / Git

-   OS version ("cat /etc/rear/os.conf" or "lsb\_release -a" or "cat
    /etc/os-release"):  
    OS\_VENDOR=RedHatEnterpriseServer  
    OS\_VERSION=7

-   ReaR configuration files ("cat /etc/rear/site.conf" and/or "cat
    /etc/rear/local.conf"):  
    OUTPUT=ISO  
    BACKUP=NETFS  
    BACKUP\_PROG=tar  
    BACKUP\_TYPE=incremental  
    FULLBACKUPDAY="Mon"  
    FULLBACKUP\_OUTDATED\_DAYS=5  
    BACKUP\_INTEGRITY\_CHECK=1  
    \#--- Backup Destination  
    BACKUP\_URL=cifs://192.168.0.95/VMs  
    BACKUP\_OPTIONS="cred=/root/.confid/.cifs"

-   Hardware (PC or PowerNV BareMetal or ARM) or virtual machine (KVM
    guest or PoverVM LPAR):  
    KVM Guest

-   System architecture (x86 compatible or PPC64/PPC64LE or what exact
    ARM device):  
    x86\_64

-   Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or
    ELILO or Petitboot):  
    BIOS and GRUB

-   Storage (local disk or SSD) and/or SAN (FC or iSCSI or FCoE) and/or
    multipath (DM or NVMe):  
    local disk

-   Description of the issue (ideally so that others can reproduce
    it):  
    According to the /var/log/rear/rear.log file the error occurs when
    rear try to mount the backup destination.

race 6: /usr/share/rear/lib/\_input-output-functions.sh:372
StopIfError  
++ echo 'Message: Mount command '''mount -v -o
cred=/root/.confid/.cifs  
//192.168.0.95/VMs  
/tmp/rear.siQhBqYx0ObcX4a/outputfs''' failed.'  
Message: Mount command 'mount -v -o cred=/root/.confid/.cifs  
//192.168.0.95/VMs  
/tmp/rear.siQhBqYx0ObcX4a/outputfs' failed.

I am able to run the command without an issue  
mount.cifs -o cred=/root/.confid/.cifs //192.168.0.95/VMs /mnt/test  
or  
mount -v -o cred=/root/.confid/.cifs //192.168.0.95/VMs /mnt/test

-   Workaround, if any:

-   Attachments, as applicable ("rear -D mkrescue/mkbackup/recover"
    debug log files):  
    [debug.log](https://github.com/rear/rear/files/3474387/debug.log)

#### <img src="https://avatars.githubusercontent.com/u/1788608?u=925fc54e2ce01551392622446ece427f51e2f0ce&v=4" width="50">[jsmeix](https://github.com/jsmeix) commented at [2019-08-07 12:05](https://github.com/rear/rear/issues/2202#issuecomment-519067145):

@chumunga  
the actual error in your 'debug.log' file is

    +++ mount -v -o $'credentials=/root/.confid/.cifs\r' $'//192.168.0.95/VMs\r' /tmp/rear.OGqvl1K61fL8dT2/outputfs
    error 2 (No such file or directory) opening credential file /root/.confid/.cifs

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-07 16:16](https://github.com/rear/rear/issues/2202#issuecomment-519167380):

Correct I saw that, however I can run the mount commanded independently
using that same credential file without an issue.

mount.cifs -o cred=/root/.confid/.cifs //192.168.0.95/VMs /mnt/test  
or  
mount -v -o cred=/root/.confid/.cifs //192.168.0.95/VMs /mnt/test

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-08 16:35](https://github.com/rear/rear/issues/2202#issuecomment-519594601):

Any ideas?

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-08 18:39](https://github.com/rear/rear/issues/2202#issuecomment-519639531):

I also tried rear -v mkbackuponly and I get a different error right,
strange (let me know if you want the debug file)  
![Annotation 2019-08-08
113903](https://user-images.githubusercontent.com/38339229/62728655-2adde780-b9d1-11e9-8fc0-0d96623ba35e.png)

#### <img src="https://avatars.githubusercontent.com/u/12116358?u=1c5ba9dcee5ca3082f03029a7fbe647efd30eb49&v=4" width="50">[gozora](https://github.com/gozora) commented at [2019-08-08 19:39](https://github.com/rear/rear/issues/2202#issuecomment-519659197):

@chumunga  
Are you sure that *local.conf* you've provided in issue description and
one you've used are one and the same file ?  
Because I can see in your debug log:

    + source /etc/rear/local.conf
    ++ OUTPUT=$'ISO\r'
    ++ BACKUP=$'NETFS\r'
    ++ BACKUP_PROG=$'tar\r'
    ++ BACKUP_TYPE=$'incremental\r'
    ++ FULLBACKUPDAY=$'Mon\r'
    ++ FULLBACKUP_OUTDATED_DAYS=$'5\r'
    ++ BACKUP_INTEGRITY_CHECK=$'1\r'
    ++ BACKUP_URL=$'cifs://192.168.0.95/VMs\r'
    ++ BACKUP_OPTIONS=$'credentials=/root/.confid/.cifs\r'

Notice the **$** and **\\r** ...

Can it be that your editor is somehow "misconfigured" ?

V.

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-09 16:13](https://github.com/rear/rear/issues/2202#issuecomment-519977493):

huh....interesting i wonder if it has something to do with the fact that
i wrote it in windows notepadd++ then transferred the file via WinSCP I
will type the file out and see what happens.

#### <img src="https://avatars.githubusercontent.com/u/12116358?u=1c5ba9dcee5ca3082f03029a7fbe647efd30eb49&v=4" width="50">[gozora](https://github.com/gozora) commented at [2019-08-09 16:29](https://github.com/rear/rear/issues/2202#issuecomment-519982242):

Well yes, this can be the problem indeed.

v.

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-09 17:46](https://github.com/rear/rear/issues/2202#issuecomment-520006148):

That was it, i manually typed out the local.conf file and the backup
works. I am glad since I have 3 more servers that is having the same
issues.

#### <img src="https://avatars.githubusercontent.com/u/1788608?u=925fc54e2ce01551392622446ece427f51e2f0ce&v=4" width="50">[jsmeix](https://github.com/jsmeix) commented at [2019-08-12 12:42](https://github.com/rear/rear/issues/2202#issuecomment-520409001):

@gozora  
thank you for taking such little details seriously.  
I had noticed that strange looking values but I thought  
it is not real but just some kind of "optical noise".

#### <img src="https://avatars.githubusercontent.com/u/38339229?v=4" width="50">[chumunga](https://github.com/chumunga) commented at [2019-08-14 18:26](https://github.com/rear/rear/issues/2202#issuecomment-521362681):

Thanks again, I thought that was just the debug file format and wasn't
literally what was being thrown at the command line.

And if anyone ever runs into a similar situation, you can use a program
called dos2unix

------------------------------------------------------------------------

\[Export of Github issue for
[rear/rear](https://github.com/rear/rear).\]
